ruma-identity-service-api/src/keys/valid_public_key/v2.rs

```diff
@@ -0,0 +1,56 @@
+//! [GET /_matrix/identity/v2/pubkey/isvalid](https://matrix.org/docs/spec/identity_service/latest#get-matrix-identity-v2-pubkey-isvalid)
+
+use ruma_api::ruma_api;
+
+ruma_api! {
+    metadata: {
+        description: "Check whether a long-term public key is valid. The response should always be the same, provided the key exists.",
+        method: GET,
+        name: "valid_public_key",
+        path: "/_matrix/identity/v2/pubkey/isvalid",
+        authentication: None,
+        rate_limited: false,
+    }
+
+    #[derive(Default)]
+    request: {}
+
+    response: {
+        /// Whether the public key is recognised and is currently valid.
+        pub valid: bool,
+    }
+}
+
+impl Request {
+    /// Create an empty `Request`.
+    ///
+    /// Note: client must supply public key query parameter.
+    ///
+    /// Example:
+    ///
+    /// ```rust-no-build
+    /// # use std::collections::BTreeMap;
+    /// #
+    /// # use http::uri::Uri;
+    /// #
+    /// # use ruma::ruma_client;
+    /// # use ruma::ruma_identity_service_api::keys::valid_public_key;
+    ///
+    /// let homeserver_uri = "https://some.matrix.server".parse::<Uri>().unwrap();
+    /// let client = ruma_client::Client::new(homeserver_uri, None);
+    /// let req = valid_public_key::v2::Request::new();
+    /// let mut params: BTreeMap<String, String> = BTreeMap::new();
+    /// params.insert("public_key", "M9TCOCDzBevee/nRK7A8JFDQbv6OHFHDkJQswpPMzvM");
+    /// let _ = client.request_with_url_params(req, Some(params));
+    /// ```
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Response {
+    /// Create a `Response` with the given parameters.
+    pub fn new(_valid: bool) -> Self {
```

Comment:
If you remove the leading underscore you can initialize the struct with field init shorthand (i.e. replace `valid: _valid` with just `valid')